drivers/wifi/siwx917/siwx917_wifi.c

```diff
@@ -197,19 +197,30 @@ static int siwx917_scan(const struct device *dev, struct wifi_scan_params *z_sca
 {
 	sl_wifi_scan_configuration_t sl_scan_config = { };
 	struct siwx917_dev *sidev = dev->data;
+	sl_wifi_ssid_t direct_ssid = {0};
```

Comment:
Since you left `sl_scan_config` as is, please be constent here with the initialization.